R/tables_spark.R

```diff
@@ -0,0 +1,73 @@
+tbl_quote_name <- function(name) {
+  x <- gsub("`", "``", name, fixed = TRUE)
+  paste("`", name, "`", sep = "")
+}
+
+tbl_cache_sdf <- function(sc, name, force) {
+  tbl <- tbl(sc, name)
+  sdf <- spark_dataframe(tbl)
+
+  invoke(sdf, "cache")
+  if (force)
+    invoke(sdf, "count")
+}
+
+tbl_cache_sql <- function(sc, name, force) {
+  sql <- paste("CACHE TABLE", tbl_quote_name(name))
+  invoke(hive_context(sc), "sql", sql)
+
+  if (force) {
+    sql <- paste("SELECT count(*) FROM ", tbl_quote_name(name))
+    sdf <- invoke(hive_context(sc), "sql", sql)
+    sdf_collect(sdf)
+  }
+}
+
+#' Cache a Spark Table
+#'
+#' Force a Spark table with name \code{name} to be loaded into memory.
+#' Operations on cached tables should normally (although not always)
+#' be more performant than the same operation performed on an uncached
+#' table.
+#'
+#' @param sc A \code{spark_connection}.
+#' @param name The table name.
+#' @param force Force the data to be loaded into memory? This is accomplished
+#'   by calling the \code{count} API on the associated Spark DataFrame.
+#'
+#' @export
+tbl_cache <- function(sc, name, force = TRUE) {
+  countColumns <- function(sc, name) {
+    sql <- paste("SELECT * FROM ", tbl_quote_name(name))
```

Comment:
This seems like it would be a potentially expensive query if all we need is the number of columns (or is Spark SQL smart enough to not materialize this expression?)